changes/224.removal.1.rst

```diff
@@ -0,0 +1 @@
+Supplying an applicator to BaseStyle.copy() has been deprecated. If you need to manually assign an applicator to a style, do it separately, after the copy. (In normal usage, though, assigning an applicator to a style is handled automatically when styles and/or applicators are assigned to nodes.)
```

Comment:
Hm, fair enough. I was just thinking in terms of Toga specifically, I think — and a Toga user should probably never encounter this anyway. And thank you for linking that issue! Didn't realize it was already being tracked.

Reply:
Removed the parenthetical, added separate bugfix changenote, and edited that issue into the initial comment for the PR.

Reply:
Oh, right. Had momentary amnesia about what you said about whether or not to mark this as fixing it. I think it makes... Toga only requires Travertino of *at least* 0.3.0, right? So as soon as this is included in a released version, it's available to current version of Toga. With the caveat that current installations aren't going to be magically auto-updated, of course...

Reply:
I suppose it boils down to a sort of false positive question. Once Travertino 0.3.1 or 0.4.0 or whatever is released with this patch, it is entirely *possible* to have an installation of Toga 0.4.8 in which the bug is fixed / feature is added. And any *new* installation of it presumably will, unless the user specifically pins Travertino 0.3.0 as part of the install.

But it won't be *guaranteed* to be fixed on existing installations, unless the user updates Travertino, or until they update to the next version of Toga, which can require the newer version of Travertino.

I'm still inclined to think this counts, but as someone without much experience in project management conventions, it's not a hill I'd argue on if you disagree.

Reply:
I agree with your analysis - this PR *will* fix both bugs in Toga... but in a way that causes warnings in Toga. The catch will be how we release this in a way that minimises disruption to users. If existing Toga users are going to get a thousand warnings when they upgrade Travertino, we may need to do a near parallel release to make sure there's a clear way to avoid the warning.

Reply:
I think that sounds like the way to go.

1. Finish #222
2. Finish https://github.com/beeware/toga/pull/2443 and https://github.com/beeware/toga/issues/2937, but don't merge them yet
3. Release a new version of Travertino
4. Merge the PRs and release a new version of Toga
5. ???
6. Profit!

Reply:
That's definitely one approach; the downside being that it will be difficult to avoid some users getting the warnings from Travertino. 

The other possible approach would be something like:

1. Land beeware/toga#2937 in a way that it works with *both* Travertino releases. 
2. Cut a Toga release
3. Complete the work on #222 
4. Cut a Travertino release
5. Complete the work on beeware/toga#2443
6. Cut a second Toga release

That all hinges on (1) being possible; but if it is, this approach would give a little breathing room between the Toga release and the Travertino release, so the chances of people getting stuck in the middle zone is lessened. It also gives a little more space for (3) and (5) to be completed - and ironically, the longer they take to implement, the less the impact of the Travertino change will be.

Reply:
Ah, that makes sense! I had assumed # 1 would be ungainly, but I tried it out just now and what do you know, it's totally doable.